pkg/openstack/loadbalancer_test.go

```diff
@@ -534,3 +535,153 @@ func Test_getListenerProtocol(t *testing.T) {
+					},
+				},
+				annotationKey:  "annotationKey",
+				defaultSetting: "default-setting",
+			},
+			expect: "default-setting",
+		},
+		{
+			name: "enter valid arguments with annotations",
+			testArgs: testArgs{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Namespace:   "service-namespace",
+						Name:        "service-name",
+						Annotations: map[string]string{"annotationKey": "annotation-Value"},
+					},
+				},
+				annotationKey:  "annotationKey",
+				defaultSetting: "default-setting",
+			},
+			expect: "annotation-Value",
+		},
+		{
+			name: "enter argument without default-setting",
+			testArgs: testArgs{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Namespace:   "service-namespace",
+						Name:        "service-name",
+						Annotations: map[string]string{"annotationKey": "annotation-Value"},
+					},
+				},
+				annotationKey:  "annotationKey",
+				defaultSetting: "",
+			},
+			expect: "annotation-Value",
+		},
+		{
+			name: "enter argument without annotation and default-setting",
+			testArgs: testArgs{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Namespace: "service-namespace",
+						Name:      "service-name",
+					},
+				},
+				annotationKey:  "annotationKey",
+				defaultSetting: "",
+			},
+			expect: "",
+		},
+		{
+			name: "enter argument with a non-existing annotationKey with default setting",
+			testArgs: testArgs{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Namespace:   "service-namespace",
+						Name:        "service-name",
+						Annotations: map[string]string{"annotationKey": "annotation-Value"},
+					},
+				},
+				annotationKey:  "invalid-annotationKey",
+				defaultSetting: "default-setting",
+			},
+			expect: "default-setting",
+		},
+		{
+			name: "enter argument with a non-existing annotationKey without a default setting",
+			testArgs: testArgs{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Namespace:   "service-namespace",
+						Name:        "service-name",
+						Annotations: map[string]string{"annotationKey": "annotation-Value"},
+					},
+				},
+				annotationKey:  "invalid-annotationKey",
+				defaultSetting: "",
+			},
+			expect: "",
+		},
+		{
+			name: "no name-space and service name but valid annotations",
+			testArgs: testArgs{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{"annotationKey": "annotation-Value"},
+					},
+				},
+				annotationKey:  "annotationKey",
+				defaultSetting: "default-setting",
+			},
+			expect: "annotation-Value",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if got := getStringFromServiceAnnotation(test.testArgs.service, test.testArgs.annotationKey, test.testArgs.defaultSetting); got != test.expect {
+				t.Errorf("[Error] : getStringFromServiceAnnotation() => %s, expecting = %s", got, test.expect)
+			}
```

Comment:
This should be done through `assert.Equal()`.

Reply:
noted! and fixed